packages/openapi-v3/src/controller-spec.ts

```diff
@@ -89,16 +90,55 @@ function resolveControllerSpec(constructor: Function): ControllerSpec {
       endpointName = `${fullMethodName} (${verb} ${path})`;
     }
 
+    const defaultResponse = {
+      '200': {
+        description: `Return value of ${constructor.name}.${op}`,
+      },
+    };
+
     let operationSpec = endpoint.spec;
     if (!operationSpec) {
       // The operation was defined via @operation(verb, path) with no spec
       operationSpec = {
-        responses: {},
+        responses: defaultResponse,
       };
       endpoint.spec = operationSpec;
     }
     debug('  operation for method %s: %j', op, endpoint);
 
+    debug('  spec responses for method %s: %o', op, operationSpec.responses);
+
+    const TS_TYPE_KEY = 'x-ts-type';
+
+    for (const code in operationSpec.responses) {
+      for (const c in operationSpec.responses[code].content) {
+        debug('  evaluating response code %s with content: %o', code, c);
+        const tsType = operationSpec.responses[code].content[c][TS_TYPE_KEY];
```

Comment:
Please extract `operationSpec.responses[code].content[c]` into a shared variable to avoid repetition.